### PR TITLE
ocrvs-3852 Added conditional parameters for custom form field 

### PR DESCRIPTION
**Pull request by sadmananik**
FARAJALAND PR: https://github.com/opencrvs/opencrvs-farajaland/pull/355

ocrvs-3852

### CODE REVIEW

File: packages/client/src/components/formConfig/formTools/components.tsx [hunk 1]
```diff
@@ -20,6 +20,19 @@ import {
...
+import { formSectionToFieldIdentifiers } from '@client/forms/questionConfig/transformers'
+import { useParams } from 'react-router'
+import { getFieldId } from '@client/forms/configuration/defaultUtils'
+import { IConditionalConfig } from '@client/forms/questionConfig'
+
+export const TitleWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 8px;
+`
```
Comment (Zangetsu101) [inline packages/client/src/components/formConfig/formTools/components.tsx:35]: This also can be replaced by `<Stack />`

File: packages/client/src/components/formConfig/formTools/CustomFieldTools.tsx [hunk 2]
```diff
@@ -413,8 +486,18 @@ class CustomFieldToolsComp extends React.Component<
     )
   }
 
-  toggleButtons() {
+  toggleButtons(
+    fieldIds: {
+      [key: string]: string
+    }[]
```
Comment (Zangetsu101) [inline packages/client/src/components/formConfig/formTools/CustomFieldTools.tsx:492]:
```suggestion
    fieldIds: IConditionalConfig[]
```

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/components/formConfig/formTools/components.tsx (hunk 1 above)
Comment (jpye-finch) [reply packages/client/src/components/formConfig/formTools/components.tsx:35]: FYI <Text /> can take an icon. 
https://ui-kit.farajaland-staging.opencrvs.org/?path=/docs/typography-text--default

Comment (sadmananik) [reply packages/client/src/components/formConfig/formTools/components.tsx:35]: Yes. Thanks for point it out.

File: packages/gateway/src/features/questions/schema.graphql [hunk 3]
```diff
@@ -30,6 +30,16 @@ input MesssageInput {
   descriptor: MesssageDescriptorInput!
 }
 
+type Conditional {
+  fieldId: String
+  regexp: String
+}
```
Comment (Zangetsu101) [inline packages/gateway/src/features/questions/schema.graphql:36]: I don't think we need these

File: packages/gateway/src/features/questions/schema.graphql [hunk 4]
```diff
@@ -30,6 +30,16 @@ input MesssageInput {
...
+
+input ConditionalInput {
+  fieldId: String
+  regexp: String
+}
```
Comment (Zangetsu101) [inline packages/gateway/src/features/questions/schema.graphql:41]:
```suggestion
input ConditionalInput {
  fieldId: String!
  regexp: String!
}
```

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**